File: TUnit.TestProject/ConsoleConcurrentTests.cs
namespace TUnit.TestProject;

public class ConsoleConcurrentTests
{
    [Test, Repeat(25)]
    public void Test1()
    {
        Console.WriteLine(nameof(Test1));
    }

    [Test, Repeat(25)]
    public void Test2()
    {
        Console.WriteLine(nameof(Test2));
    }

    [Test, Repeat(25)]
    public void Test3()
    {
        Console.WriteLine(nameof(Test3));
    }

    [Test, Repeat(25)]
    public void Test4()
    {
        Console.WriteLine(nameof(Test4));
    }

    [Test, Repeat(25)]
    public void Test5()
    {
        Console.WriteLine(nameof(Test5));
    }

    [Test, Repeat(25)]
    public void Test6()
    {
        Console.WriteLine(nameof(Test6));
    }

    [Test, Repeat(25)]
    public void Test7()
    {
        Console.WriteLine(nameof(Test7));
        Console.WriteLine(nameof(Test7));
        Console.WriteLine(nameof(Test7));
        Console.WriteLine(nameof(Test7));
        Console.WriteLine(nameof(Test7));
    }

    [Test, Repeat(25)]
    public void Test8()
    {
        Console.WriteLine(nameof(Test8));
    }

    [Test, Repeat(25)]
    public void Test9()
    {
        Console.WriteLine(nameof(Test9));
    }

    [Test, Repeat(25)]
    public void Test10()
    {
        Console.WriteLine(nameof(Test10));
    }
}

File: TUnit.TestProject.FSharp/ConsoleConcurrentTests.fs
namespace TUnit.TestProject.FSharp

open System
open TUnit.Core

// F# equivalent of ConsoleConcurrentTests.cs

type ConsoleConcurrentTests() =
    [<Test>]
    [<Repeat(25)>]
    member _.Test1() = Console.WriteLine("Test1")
    [<Test>]
    [<Repeat(25)>]
    member _.Test2() = Console.WriteLine("Test2")
    [<Test>]
    [<Repeat(25)>]
    member _.Test3() = Console.WriteLine("Test3")
    [<Test>]
    [<Repeat(25)>]
    member _.Test4() = Console.WriteLine("Test4")
    [<Test>]
    [<Repeat(25)>]
    member _.Test5() = Console.WriteLine("Test5")
    [<Test>]
    [<Repeat(25)>]
    member _.Test6() = Console.WriteLine("Test6")
    [<Test>]
    [<Repeat(25)>]
    member _.Test7() =
        Console.WriteLine("Test7")
        Console.WriteLine("Test7")
        Console.WriteLine("Test7")
        Console.WriteLine("Test7")
        Console.WriteLine("Test7")
    [<Test>]
    [<Repeat(25)>]
    member _.Test8() = Console.WriteLine("Test8")
    [<Test>]
    [<Repeat(25)>]
    member _.Test9() = Console.WriteLine("Test9")
    [<Test>]
    [<Repeat(25)>]
    member _.Test10() = Console.WriteLine("Test10")
